src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,45 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.DukeException;
+
+
+
+/**
+ * Represents a parser object that will read the input from
+ * the users and then parse the input to the program.
+ */
+public class Parser {
+    /**
+     * The function will take the user message and then parse it to return the command objects
+     * for further procedures.
+     *
+     * @param userMessage The message the user inputs.
+     * @throws DukeException if the user input message can't be parsed.
+     * @return a specific subclass of the Command object. eg. ListCommand object.
```

Comment:
According to https://se-education.org/guides/conventions/java/basic.html#comments, please state @return before @throws